[BUG] Invalid negative data sent: missing integer in path

### Checklist

- [x] I checked the [FAQ section](https://schemathesis.readthedocs.io/en/stable/faq/) of the documentation
- [x] I looked for similar issues in the [issue tracker](https://github.com/schemathesis/schemathesis/issues)
- [x] I am using the latest version of Schemathesis

### Describe the bug

Schemathesis sends invalid negative data: the resource tested requires an integer in path, and the request is sent without this integer.

Note: this seems to be a regression in v4.0.x as tests were OK with schemathesis v3.x.

Consequence: the URL sent matches another resource; schemathesis expects an error, but the URL is valid and the other schema is returned, which leads to an error displayed by schemathesis:

```
======================================================================================= FAILURES ======================================================================================
_______________________________________________________________________________ GET /buffers/{buffer_id} ______________________________________________________________________________
1. Test Case ID: 1kOfz4

- Response violates schema

    [{'id': 1750742411684629, 'name': 'core.weechat', 'short_name': 'weechat', 'number': 1, 'type': 'formatted', 'hidden': False, 'title': 'WeeChat 4.7.0-dev (C) 2003-2025 - https://weechat.org/', 'modes': '', 'input_prompt': '', 'input': '', 'input_position': 0, 'input_multiline': False, 'nicklist': False, 'nicklist_case_sensitive': False, 'nicklist_display_groups': True, 'time_displayed': True, 'local_variables': {'plugin': 'core', 'name': 'weechat'}, 'keys': [], 'nicklist_root': {'id': 0, 'parent_group_id': -1, 'name': 'root', 'color_name': '', 'color': '', 'visible': False, 'groups': [], 'nicks': []}}] is not of type 'object'

    Schema:

        {
            "type": "object",
            "properties": {
                "id": {
                    "type": "integer",
                    "format": "int64",
                    "description": "buffer identifier (\u2265 0)",
                    "example": 1709932823238637
                },
                "name": {
                    "type": "string",
                    "example": "irc.libera.#weechat"
                },
                "short_name": {
                    "type": "string",
                    "example": "#weechat"
                },
                "number": {
                    "type": "integer",
            // Output truncated...
        }

    Value:

        [
            {
                "id": 1750742411684629,
                "name": "core.weechat",
                "short_name": "weechat",
                "number": 1,
                "type": "formatted",
                "hidden": false,
                "title": "WeeChat 4.7.0-dev (C) 2003-2025 - https://weechat.org/",
                "modes": "",
                "input_prompt": "",
                "input": "",
                "input_position": 0,
                "input_multiline": false,
                "nicklist": false,
                "nicklist_case_sensitive": false,
                "nicklist_display_groups": true,
                "time_displayed": true,
                "local_variables": {
            // Output truncated...
        ]

- Accepted negative data

    Allowed statuses: 400, 401, 403, 404, 406, 422, 428, 5xx

[200] OK:

    `[{"id":1750742411684629,"name":"core.weechat","short_name":"weechat","number":1,"type":"formatted","hidden":false,"title":"WeeChat 4.7.0-dev (C) 2003-2025 - https://weechat.org/","modes":"","input_prompt":"","input":"","input_position":0,"input_multiline":false,"nicklist":false,"nicklist_case_sensitive":false,"nicklist_display_groups":true,"time_displayed":true,"local_variables":{"plugin":"core","name":"weechat"},"keys":[],"nicklist_root":{"id":0,"parent_group_id":-1,"name":"root","color_name":"","color":"" // Output truncated...`

Reproduce with: 

    curl -X GET -H 'x-weechat-totp: 0' -H 'Authorization: [Filtered]' 'http://localhost:9000/api/buffers/?lines=0&nicks=true&colors=ansi'
```

### To Reproduce

🚨 **Mandatory** 🚨: Steps to reproduce the behavior:

1. Clone the weechat repository: https://github.com/weechat/weechat
2. Switch to branch: `git switch ci-schemathesis-v4`
3. Compile WeeChat and run with this command:

```
weechat --dir /tmp/weechat-test-api --run-command '/set relay.network.password "test"' --run-command '/set relay.network.password_hash_iterations 100' --run-command '/set relay.network.max_clients 0' --run-command '/relay add api 9000'
```

4. Run schemathesis in another terminal, in root of git repository:

```
RELAY_PASSWORD=test schemathesis run --phases coverage --url http://localhost:9000/api src/plugins/relay/api/weechat-relay-api.yaml
```

Schema: https://github.com/weechat/weechat/blob/ci-schemathesis-v4/src/plugins/relay/api/weechat-relay-api.yaml

### Expected behavior

No error from schemathesis.

### Environment

```
- OS: Linux Debian Sid
- Python version: 3.13.5
- Schemathesis version: 4.0.2
- Spec version: Open API 3.1.0
```

### Additional context

The path parameter `buffer_id` is defined like this:

```yaml
  parameters:
    bufferId:
      name: buffer_id
      in: path
      required: true
      schema:
        type: integer
        format: int64
      description: Buffer identifier
      example: 1709932823238637
```

The "negative data" sent by schemathesis sends an empty value in URL which is not valid, because this parameter is "required".
It then matches another resource `/buffers` without parameters, which is a list of buffers and not a single buffer, causing the error displayed by schemathesis.

CI example showing the issue: https://github.com/weechat/weechat/actions/runs/15835220404/job/44637128820

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -2,6 +2,10 @@
 
 ## [Unreleased](https://github.com/schemathesis/schemathesis/compare/v4.0.3...HEAD) - TBD
 
+### :bug: Fixed
+
+- Empty path parameter in the negative tests. [#2912](https://github.com/schemathesis/schemathesis/issues/2912)
+
 ## [4.0.3](https://github.com/schemathesis/schemathesis/compare/v4.0.2...v4.0.3) - 2025-06-30
 
 ### :wrench: Changed

--- a/src/schemathesis/generation/coverage.py
+++ b/src/schemathesis/generation/coverage.py
@@ -1060,7 +1060,7 @@ def _negative_type(
         strategies["number"] = FLOAT_STRATEGY.filter(_is_non_integer_float)
     for strategy in strategies.values():
         value = ctx.generate_from(strategy)
-        if seen.insert(value):
+        if seen.insert(value) and ctx.is_valid_for_location(value):
             yield NegativeValue(value, description="Incorrect type", location=ctx.current_path)
 
 

--- a/test/coverage/test_phase.py
+++ b/test/coverage/test_phase.py
@@ -1646,6 +1646,55 @@ def test_query_parameters_dont_exceed_max_length(ctx):
     )
 
 
+def test_path_parameters_always_present(ctx):
+    schema = ctx.openapi.build_schema(
+        {
+            "/foo/{foo_id}": {
+                "post": {
+                    "parameters": [
+                        {
+                            "name": "foo_id",
+                            "in": "path",
+                            "required": True,
+                            "schema": {
+                                "type": "integer",
+                            },
+                        },
+                    ],
+                    "responses": {"default": {"description": "OK"}},
+                }
+            },
+        }
+    )
+    assert_coverage(
+        schema,
+        [GenerationMode.NEGATIVE],
+        [
+            {
+                "path_parameters": {
+                    "foo_id": {},
+                },
+            },
+            {
+                "path_parameters": {
+                    "foo_id": "null,null",
+                },
+            },
+            {
+                "path_parameters": {
+                    "foo_id": "null",
+                },
+            },
+            {
+                "path_parameters": {
+                    "foo_id": "false",
+                },
+            },
+        ],
+        ("/foo/{foo_id}", "post"),
+    )
+
+
 def test_negative_query_parameter(ctx):
     schema = ctx.openapi.build_schema(
         {

